.github/workflows/release-waypoint.yaml

```diff
@@ -0,0 +1,54 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-22.04
+            arch: x86_64
+            binary_suffix: '-linux-x86_64'
+          - os: macos-latest
+            arch: x86_64
+            binary_suffix: '-macos-x86_64'
+          - os: macos-latest
+            arch: arm64
+            binary_suffix: '-macos-arm64'
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        run: |
+          pip install pyinstaller
```

Comment:
nit: u can use `working-directory: docker/waypoint` here instead of cd, but dont have to address